universal-application-tool-0.0.1/test/controllers/admin/ProgramAdminManagementControllerTest.java

```diff
@@ -90,9 +90,42 @@ public void update_succeeds() {
     assertThat(account2.getAdministeredProgramNames()).containsOnly(programName);
   }
 
+  @Test
+  public void update_withRemovals_succeeds() {
+    String programName = "add remove";
+    Program program = ProgramBuilder.newDraftProgram(programName).build();
+
+    String addEmail = "add me";
+    Account addAccount = new Account();
+    addAccount.setEmailAddress(addEmail);
+    addAccount.save();
+
+    String removeEmail = "remove me";
+    Account removeAccount = new Account();
+    removeAccount.setEmailAddress(removeEmail);
+    removeAccount.addAdministeredProgram(program.getProgramDefinition());
+    removeAccount.save();
+
+    Http.Request request =
+        fakeRequest()
```

Comment:
Thoughts (non-blocking): why does the request not require CSRF token?

Reply:
Great question! I asked @ndmckinley and they said:

> When you call controller.update  with the request, you bypass the filter system (which happens before a request hits the controller).  the csrf protection is implemented as a filter